x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_stops.js

```diff
@@ -87,7 +96,10 @@ export const ColorStops = ({
             defaultMessage: 'Color must provide a valid hex value',
           })
         : undefined,
-      colorInput: <EuiColorPicker onChange={onColorChange} color={color} compressed />,
+      colorInput: {
```

Comment:
The refactoring of `getColorInput` makes it so that the method no longer returns an input at all. In fact, `colorInput` just echos the input parameters. Would you mind just removing the method `getColorInput` entirely?

So instead of 
```
const { colorError, colorInput } = getColorInput(onColorChange, colorStop.color);
```

You would just have 
```
colorError: isColorInvalid(colorStop.color)
        ? i18n.translate('xpack.maps.styles.colorStops.hexWarningLabel', {
            defaultMessage: 'Color must provide a valid hex value',
          })
        : undefined,
```